specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json

```diff
@@ -2424,38 +2424,10 @@
           },
           "apiVersionSet" : {
             "description": "Version set details",
-            "$ref" : "#/definitions/ApiVersionSetContract"
+            "$ref" : "./apimversionsets.json#/definitions/ApiVersionSetContractProperties"
```

Comment:
Note that changing the type of the property `apiVersionSet` in this model is breaking change. 

If we take the diff of `ApiVersionSetContract` (old type) and `ApiVersionSetContractProperties` (new type), it looks like the read-only properties `id`, `name` and `type` won't be available for consumer any more. 

Changing the type is a SDK breaking change. This requires major SDK version upgrade.

Removing properties in the response of a stable api call (without bumping api version) is service side breaking change, which requires ARM approval.

Reply:
@anuchandy  This isn't breaking a change to the current API, this is fixing an incorrect description.  Customer's who attempt to update those fields in the GA release will not be able to.

In the preview release of this API, the customers could create ApiVersionSet directly when creating the API, but it was decided that the ARM guidelines prevented this from being allowed.  Therefore we had to remove this functionality for our customers in our GA release.  The change to make the fields read-only was missed when doing the update.   Our customers are now waiting for the documentation to be updated so they can use the new ARM compliant way of creating these things in a two step process.

If you could unblock this PR so that we can get that documentation to the customer, it would be most appreciated.

/cc @ravbhatnagar

Reply:
Also we need ARM review because there is a new swagger `apimversionsets.json`, this is the first PR adding this swagger. So as per the established process, it is required to have this sign-ed off by ARM in this repo. I'm sending an email thread requesting ARM to take a look.

Reply:
@anuchandy - I think what @darrelmiller is saying is that at GA, customer could not update/assign the value of id, name, type since they were read only. Swagger didnt reflect it. And so now, swagger is updating the description correctly to reflect the service state. If true, this is not a breaking change, since it would have never worked for the customer even if he tried to update.

Reply:
@anuchandy The current SDK is from March 2017 and is a preview https://www.nuget.org/packages/Microsoft.Azure.Management.ApiManagement/3.4.0-preview

Reply:
@darrelmiller thanks for letting me know. Based on the process we established,  i'll add `potential SDK breaking changes` tag for future reference. 

Once we conclude the below discussion with ARM team, will merge this PR.